substrate/frame/transaction-payment/src/payment.rs

```diff
@@ -66,18 +70,95 @@ pub trait OnChargeTransaction<T: Config> {
 	) -> Result<(), TransactionValidityError>;
 }
 
-/// Implements the transaction payment for a pallet implementing the `Currency`
+/// Implements transaction payment for a pallet implementing the [`frame_support::traits::fungible`]
+/// trait (eg. pallet_balances) using an unbalance handler (implementing
+/// [`OnUnbalanced`]).
+///
+/// The unbalance handler is given 2 unbalanceds in [`OnUnbalanced::on_unbalanceds`]: `fee` and
+/// then `tip`.
+pub struct FungibleAdapter<F, OU>(PhantomData<(F, OU)>);
```

Comment:
I assume this new type is tested with previously existing tests?

Reply:
Yep, already caught an issue with the existing tests